terraform/modules/tf_threat_intel_downloader/iam.tf

```diff
@@ -0,0 +1,144 @@
+// IAM Role: Execution Role
+resource "aws_iam_role" "threat_intel_downloader" {
+  name               = "${var.prefix}_threat_intel_downloader"
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: Generic Lambda AssumeRole
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Role Policy: Allow lambda function to invoke the Lambda Function
+resource "aws_iam_role_policy" "threat_intel_downloader" {
+  name   = "invoke_lambda_function_role_policy"
+  role   = "${aws_iam_role.threat_intel_downloader.id}"
+  policy = "${data.aws_iam_policy_document.invoke_lambda_function.json}"
+}
+
+// IAM Policy Doc: Allow the lambda function to invoke the Lambda function
+data "aws_iam_policy_document" "invoke_lambda_function" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "lambda:InvokeFunction",
+    ]
+
+    resources = [
+      "arn:aws:lambda:${var.region}:${var.account_id}:function:${var.prefix}_streamalert_threat_intel_downloader",
+    ]
+  }
+}
+
+// IAM Role Policy: Allow the lambda function to create/update CloudWatch logs
+resource "aws_iam_role_policy" "cloudwatch_logs" {
+  name   = "cloudwatch_logs_role_policy"
+  role   = "${aws_iam_role.threat_intel_downloader.id}"
+  policy = "${data.aws_iam_policy_document.cloudwatch_logs_policy.json}"
+}
+
+// IAM Policy Doc: Allow the lambda function to create/update CloudWatch logs
+data "aws_iam_policy_document" "cloudwatch_logs_policy" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+    ]
+
+    resources = [
+      "${aws_cloudwatch_log_group.threat_intel_downloader.arn}",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = [
+      "arn:aws:logs:${var.region}:${var.account_id}:log-group:${aws_cloudwatch_log_group.threat_intel_downloader.name}:log-stream:*",
+    ]
+  }
+}
+
+// IAM role policy: Allow lambda function to read/write data from DynamoDB
+resource "aws_iam_role_policy" "read_write_dynamodb" {
+  name   = "read_dynamodb"
+  role   = "${aws_iam_role.threat_intel_downloader.id}"
+  policy = "${data.aws_iam_policy_document.read_write_dynamodb.json}"
+}
+
+// IAM Policy Doc: Allow lambda function to read/write data from DynamoDB
+data "aws_iam_policy_document" "read_write_dynamodb" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "dynamodb:BatchGetItem",
+      "dynamodb:GetItem",
+      "dynamodb:BatchWriteItem",
+      "dynamodb:PutItem",
+    ]
+
+    resources = [
+      "arn:aws:dynamodb:${var.region}:${var.account_id}:table/${var.prefix}_streamalert_threat_intel_downloader",
```

Comment:
can this be interpolated?

Reply:
Can you have more content? Do you mean interpolated in different way?